src/Core/Product/ProductPresenter.php

```diff
@@ -823,4 +828,27 @@ protected function getProductEmbeddedAttributes(array $product)
 
         return $embeddedProductAttributes;
     }
+
+    /**
+     * Assemble the same features in one array
+     * @param  array $presentedProduct
+     * @return array
+     */
+    public function addFeaturesToDisplay($presentedProduct)
```

Comment:
Can you please make this function protected? And add the type before the variable if we MUST send an array?

```php
protected function addFeaturesToDisplay(array $presentedProduct)
```

We can also going further, by not sending the whole $presentedProduct variable. You just need the ['features'] attribute, so send it alone (Which is even accurate when you check it before calling the function).